x-pack/elastic-agent/spec/fleet-server.yml

```diff
@@ -0,0 +1,61 @@
+name: Fleet Server
+cmd: fleet-server
+args: ["--agent-mode"]
+artifact: fleet-server
+rules:
+  - fix_stream: {}
+
+  - filter_values:
+      selector: inputs
+      key: type
+      values:
+        - fleet-server
+
+  - filter_values:
+      selector: inputs
+      key: enabled
+      values:
+        - true
+
+  - remove_key:
+      key: output
+
+  - select_into:
+      selectors: [ fleet.server.output.elasticsearch ]
+      path: output
+
+  - map:
+      path: fleet
+      rules:
+        - remove_key:
+            key: enabled
+        - remove_key:
+            key: access_api_key
+        - remove_key:
+            key: kibana
+        - remove_key:
+            key: reporting
+        - remove_key:
+            key: server
+
+  - map:
```

Comment:
In general I would expect this spec file to only require the input routing to fleet-server. Why is this needed? Is this something we just add for every input automatically and because of this need to remove it?

Reply:
Yes every input automatically gets these items from Elastic Agent, so they need to be removed.